Fix capsule flashing again

SergiiDmytruk opened the pull request:
Recent https://github.com/Dasharo/coreboot/pull/686 caused trouble (primarily) for chips less than 32 MiB while fixing flash access beyond BIOS region.  We also need to take FUM into account for capsules, so they can write anywhere (similar change in EDK: https://github.com/Dasharo/edk2/pull/257).

*ref: prot-1829*

SergiiDmytruk committed:
intel/fast_spi/mmap_boot.c: don't add empty window

When flash chip fully fits into the main mmap window, the code must not
create an empty window backed up by RW device.  Something seems to hang,
likely because this use case was never considered by xlate code.

Change-Id: I925c1f9a1024e702ba0048037d4ea2f822488be1
Upstream-Status: Pending
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
intel/fast_spi/fast_spi.c: extend BIOS to fill fixed window

Platforms with 16 MiB chips don't start BIOS at offset 0, making it
possible to fit entire chip into the fixed memory mapped window.

Upstream-Status: Pending
Change-Id: Iddb819f20906eaa392caf2cfc63aebf8354e2049
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

krystian-hebel reviewed (dismissed):
Not merging yet, I have to review edk2 part first.

SergiiDmytruk commented:
Updated with fixed FastSPI changes, both of which were wrong and effectively undid https://github.com/Dasharo/coreboot/pull/686.

SergiiDmytruk committed:
drivers/efi: require FUM for capsules

Capsules were implemented with the assumption that FUM is available,
enforce the assumption.

Change-Id: I9a1bc73a425f66aed5490f2c5bf3d067dfb505a1
Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
